cloudimg/ms_azure.py

```diff
@@ -406,8 +416,17 @@ def upload_to_container(self, image_path, container_name, object_name,
         if image_path.startswith("http:") or image_path.startswith("https:"):
             blob_client.upload_blob_from_url(source_url=image_path, **kwargs)
         else:
-            with open(image_path, "rb") as data:
-                blob_client.upload_blob(data=data, **kwargs)
+            with open_func(image_path, "rb") as data:
+                # we need to pass the lenght to upload_blob for optimizaion and
+                # because without it the LZMA won't decompress correctly.
```

Comment:
LZMA can decompress it correctly but `blob_client.upload_blob` try to guess
length it it is None and that guess does not work well with lzma.LZMAFile

Reply:
I've changed it, can you please let me know if it's good now?

Reply:
lgtm